render/src/lib/views/sidebar.vertical/port.available/components.ts

```diff
@@ -0,0 +1,344 @@
+        }
+        return read;
+    }
+
+    private _createChart() {
+        this._ctx = this.canvas.nativeElement.getContext('2d');
+        const cSessionPort = this._getSessionPort();
+        if (cSessionPort) {
+            const cSparklineData = cSessionPort.sparkline_data;
+            this._chart = new Chart(this._ctx, {
+                type: 'line',
+                data: {
+                    labels: this._chart_labels.slice(0, Service.chart_limit),
+                    datasets: [{
+                        data: cSparklineData.slice(cSparklineData.length - Service.chart_limit),
+                        borderColor: Service.getPortColor(this.port.path),
+                        pointRadius: 0,
+                        fill: false,
+                    }]
+                },
+                options: {
+                    maintainAspectRatio: false,
+                    animation: {
+                        duration: 0
+                    },
+                    scales: {
+                        xAxes: [{
+                            display: false,
+                        }],
+                        yAxes: [{
+                            display: false,
+                            stacked: true,
+                            ticks: {
+                                beginAtZero: true,
+                            },
+                            gridLines: {
+                                drawOnChartArea: false
+                            }
+                        }]
+                    },
+                    legend: {
+                        display: false
+                    },
+                    tooltips: {
+                        enabled: false
+                    }
+                }
+            });
+        } else {
+            console.error('Something went wrong with the SessionPort entry');
+        }
+    }
+
+    private _getSessionPort(): IPort {
+        const cSessionPort = Service.sessionPort[this._session];
+        if (cSessionPort && cSessionPort[this.port.path]) {
+            return cSessionPort[this.port.path];
+        }
+        return null;
+    }
+
+    private _update() {
+        if (this._destroyed) {
+            return;
+        }
+        if (this._chart) {
+            const cSessionPort = this._getSessionPort();
+            if (cSessionPort) {
+                cSessionPort.read += this._read;
+                cSessionPort.sparkline_data.pop();
+                cSessionPort.sparkline_data.unshift(this._read);
+            } else {
+                console.error('Something went wrong with the SessionPort entry');
+            }
+
+            const cData = this._chart.config.data.datasets[0].data;
+            cData.pop();
+            cData.unshift(this._read);
+
+            this._read = 0;
+            this._chart.update();
+        }
+    }
+
+    public _ng_read(): string {
+        return this._formatLoad(Service.sessionPort[this._session][this.port.path].read);
```

Comment:
```
Service.sessionPort[this._session][this.port.path].read
```
I would say in general you have it in really many places. I mean such kind of chains. But it's dangerous. To have here success, we should be sure:
```
Service.sessionPort[this._session] !== undefined
Service.sessionPort[this._session][this.port.path] !== undefined
```
If something will go wrong (but it will) you will get here exception and plugin will stop to work.

In general, I think it isn't a good idea:
1) use an object as storage for data
2) use public property of service to have access to data